packages/flutter_tools/lib/src/migrate/migrate_config.dart

```diff
@@ -0,0 +1,257 @@
+    final File file = getFileFromPlatform(platform, projectDirectory: projectDirectory);
+    file.createSync(recursive: true);
+    String unmanagedFilesString = '';
+    for (final String path in unmanagedFiles) {
+      unmanagedFilesString += "  - '$path'\n";
+    }
+    file.writeAsStringSync('''
+# Generated section.
+platform: '$platform'
+createRevision: ${createRevision == null ? 'null' : "'$createRevision'"}
+baseRevision: ${baseRevision == null ? 'null' : "'$baseRevision'"}
+
+# User provided section
+
+# List of Local paths (relative to this file) that should be
+# ignored by the migrate tool.
+#
+# Files that are not part of the templates will be ignored by default.
+unmanagedFiles:
+$unmanagedFilesString
+''',
+    flush: true);
+  }
+
+  /// Returns the absolute path of the platform directory this config belongs to.
+  ///
+  /// For example, if the config is the android config with a project directory of `/project`,
+  /// the base path returned would be `/project/android`.
+  String getBasePath(Directory? projectDirectory) {
+    return getFileFromPlatform(platform, projectDirectory: projectDirectory).parent.absolute.path;
+  }
+
+  /// Searches the flutter project for all .migrate_config files.
+  ///
+  /// Optionally, missing files can be initialized with default values.
+  static Future<List<MigrateConfig>> parseOrCreateMigrateConfigs({List<String>? platforms, Directory? projectDirectory, String? currentRevision, String? createRevision, bool create = true}) async {
+    platforms ??= getSupportedPlatforms(includeRoot: true, flutterProject: projectDirectory == null ? null : FlutterProject.fromDirectory(projectDirectory));
+    final List<MigrateConfig> configs = <MigrateConfig>[];
+    for (final String platform in platforms) {
+      if (MigrateConfig.getFileFromPlatform(platform, projectDirectory: projectDirectory).existsSync()) {
+        // Existing config. Parsing.
+        configs.add(MigrateConfig.fromFile(getFileFromPlatform(platform, projectDirectory: projectDirectory)));
+      } else {
+        // No config found, creating empty config.
+        final MigrateConfig newConfig = MigrateConfig(
+          platform: platform,
+          createRevision: createRevision,
+          baseRevision: currentRevision,
+          unmanagedFiles: _kDefaultUnmanagedFiles[platform] != null ? _kDefaultUnmanagedFiles[platform]! : <String>[],
+        );
+        if (create) {
+          newConfig.writeFile(projectDirectory: projectDirectory);
+        }
+        configs.add(newConfig);
+      }
+    }
+    return configs;
+  }
+
+  /// Returns the File that the migrate config belongs given a platform and a project directory.
+  static File getFileFromPlatform(String? platform, {Directory? projectDirectory}) {
+    Directory? platformDir;
+    final FlutterProject project = projectDirectory == null ? FlutterProject.current() : FlutterProject.fromDirectory(projectDirectory);
+    switch (platform) {
+      case 'root': {
```

Comment:
an enum may be safer here

Reply:
I might go through the whole tools repo and unify references to the different platforms into a single enum. We currently have a few different ways of id-ing platforms already.

Reply:
I'd prefer an enum; if not, it doesn't seem like the `String? platform` arg should be nullable?